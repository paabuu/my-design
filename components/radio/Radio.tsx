import * as React from 'react';
import * as classnames from 'classnames';
import RadioGroup, { RadioContext } from './RadioGroup';

import './style/index.less';

export interface BaseRadioProps {
    checked?: boolean;
    value?: any;
    disabled?: boolean;
    size?: string;
}

type RadioProps = BaseRadioProps & React.HTMLAttributes<HTMLLabelElement>;

export default class Radio extends React.Component<RadioProps> {
    static contextType = RadioContext;
    static Group = RadioGroup;

    handleRadioCheckedChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        if (this.props.disabled) return;
        this.context.onRadioChange(e);
    }

    render() {
        const { value, checked, disabled, size, ...rest } = this.props;

        const isChecked = ('checked' in this.props) ? checked : this.context.value === value;
        const isDisabled = ('disabled' in this.props && disabled === undefined) || disabled;
        const classNames = classnames('md-radio', `md-radio-${size}`, {
            'md-radio-checked': isChecked,
            'md-radio-disabled': isDisabled
        });

        return (
            <label className="md-radio-wrapper" {...rest}>
                <span className={classNames}>
                    <input type="radio" value={value} checked={isChecked} onChange={this.handleRadioCheckedChange} />
                    <span className="md-radio-inner"></span>
                </span>
                <span style={{ cursor: disabled ? 'not-allowed' : 'pointer' }}>{this.props.children}</span>
            </label>
        );
    }
}