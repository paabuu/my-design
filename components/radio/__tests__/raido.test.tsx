import * as React from 'react';
import { render } from 'enzyme';
import Radio from '../Radio';

const { Group } = Radio;

describe('Radio Component', () => {
    it('should render correctly', () => {
        const wrapper = render(
            <Group value="r2">
                <Radio value="r1">Radio1</Radio>
                <Radio value="r2">Radio2</Radio>
            </Group>
        );
        
        expect(wrapper).toMatchSnapshot();
    });

    it('should render size props correctly', () => {
        const wrapper1 = render(
            <Group value="r2" size="small">
                <Radio value="r1">Radio1</Radio>
                <Radio value="r2">Radio2</Radio>
            </Group>
        );

        expect(wrapper1).toMatchSnapshot();

        const wrapper2 = render(
            <Group value="r2" size="small">
                <Radio value="r1">Radio1</Radio>
                <Radio value="r2">Radio2</Radio>
            </Group>
        );

        expect(wrapper2).toMatchSnapshot();
    });

    it('should render disabled props correctly', () => {
        const wrapper = render(
            <Group value="r2" disabled>
                <Radio value="r1">Radio1</Radio>
                <Radio value="r2">Radio2</Radio>
            </Group>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
