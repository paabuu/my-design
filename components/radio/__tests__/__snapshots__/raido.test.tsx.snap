// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component should render correctly 1`] = `
<div
  class="md-radio-group"
>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-default"
    >
      <input
        type="radio"
        value="r1"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:pointer"
    >
      Radio1
    </span>
  </label>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-default md-radio-checked"
    >
      <input
        checked=""
        type="radio"
        value="r2"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:pointer"
    >
      Radio2
    </span>
  </label>
</div>
`;

exports[`Radio Component should render disabled props correctly 1`] = `
<div
  class="md-radio-group"
>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-default md-radio-disabled"
    >
      <input
        type="radio"
        value="r1"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:not-allowed"
    >
      Radio1
    </span>
  </label>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-default md-radio-checked md-radio-disabled"
    >
      <input
        checked=""
        type="radio"
        value="r2"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:not-allowed"
    >
      Radio2
    </span>
  </label>
</div>
`;

exports[`Radio Component should render size props correctly 1`] = `
<div
  class="md-radio-group"
>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-small"
    >
      <input
        type="radio"
        value="r1"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:pointer"
    >
      Radio1
    </span>
  </label>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-small md-radio-checked"
    >
      <input
        checked=""
        type="radio"
        value="r2"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:pointer"
    >
      Radio2
    </span>
  </label>
</div>
`;

exports[`Radio Component should render size props correctly 2`] = `
<div
  class="md-radio-group"
>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-small"
    >
      <input
        type="radio"
        value="r1"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:pointer"
    >
      Radio1
    </span>
  </label>
  <label
    class="md-radio-wrapper"
  >
    <span
      class="md-radio md-radio-small md-radio-checked"
    >
      <input
        checked=""
        type="radio"
        value="r2"
      />
      <span
        class="md-radio-inner"
      />
    </span>
    <span
      style="cursor:pointer"
    >
      Radio2
    </span>
  </label>
</div>
`;
